deepchecks/vision/suite.py

```diff
@@ -158,21 +164,29 @@ def _update_loop(
         self,
         context: Context,
         run_train_test_checks: bool,
+        run_single_dataset: Optional[DatasetKind],
         results: Dict[Union[str, int], BaseCheckResult],
         dataset_kind: DatasetKind,
         progressbar_factory: ProgressBarGroup
     ):
         type_suffix = ' - Test Dataset' if dataset_kind == DatasetKind.TEST else ' - Train Dataset'
         vision_data = context.get_data_by_kind(dataset_kind)
+
         single_dataset_checks = {k: check for k, check in self.checks.items() if isinstance(check, SingleDatasetCheck)}
 
+        flag_skip_single = run_single_dataset not in [None, dataset_kind]
+
         # SingleDatasetChecks have different handling, need to initialize them here (to have them ready for different
         # dataset kind)
+
         for idx, check in single_dataset_checks.items():
-            try:
-                check.initialize_run(context, dataset_kind=dataset_kind)
-            except Exception as exp:
-                results[idx] = CheckFailure(check, exp, type_suffix)
+            if flag_skip_single:
+                results[idx] = -1
```

Comment:
Is this necessary? if not possible to do it in another way then  would add a comment in the code why this is the case. (we will not remember)